src/php/eZ/Publish/Profiler/Executor.php

```diff
@@ -22,6 +22,10 @@ public function run( array $constraints, Aborter $aborter )
             {
                 $constraint->run( $this, $this->logger );
             }
+
+            // Force garbage collection between runs so we do not hit it while
+            // running an executor which will fail the performance analysis.
+            gc_collect_cycles();
```

Comment:
This is not necessarily true, if the garbage collector cannot clean up the object references because of cycles, then cleaning it here explicitly will have no benefit and still trigger it within the next run. Better would be to entirely disable the garbage collection in each run, then enable after the run, trigger the cycle and disable it again.